packages/eservice-template-process/src/services/eserviceTemplateService.ts

```diff
@@ -438,6 +439,99 @@ export function eserviceTemplateServiceBuilder(
 
       return updatedEServiceTemplate;
     },
+    async patchUpdateDraftTemplateVersion(
+      eserviceTemplateId: EServiceTemplateId,
+      eserviceTemplateVersionId: EServiceTemplateVersionId,
+      seed: Partial<{
+        description: string;
+        voucherLifespan: number;
+        dailyCallsPerConsumer: number;
+        dailyCallsTotal: number;
+        agreementApprovalPolicy: "AUTOMATIC" | "MANUAL";
+      }>,
+      {
+        authData,
+        correlationId,
+        logger,
+      }: WithLogger<AppContext<UIAuthData | M2MAdminAuthData>>
+    ): Promise<WithMetadata<EServiceTemplate>> {
+      logger.info(
+        `Patch update draft e-service template version ${eserviceTemplateVersionId} for EService template ${eserviceTemplateId}`
+      );
+
+      const eserviceTemplate = await retrieveEServiceTemplate(
+        eserviceTemplateId,
+        readModelService
+      );
+
+      assertRequesterEServiceTemplateCreator(
+        eserviceTemplate.data.creatorId,
+        authData
+      );
+
+      const eserviceTemplateVersion = retrieveEServiceTemplateVersion(
+        eserviceTemplateVersionId,
+        eserviceTemplate.data
+      );
+
+      if (
+        eserviceTemplateVersion.state !== eserviceTemplateVersionState.draft
+      ) {
+        throw notValidEServiceTemplateVersionState(
+          eserviceTemplateVersionId,
+          eserviceTemplateVersion.state
+        );
+      }
+
+      // Only check consistency if both dailyCallsPerConsumer and dailyCallsTotal are provided
+      if (
+        seed.dailyCallsPerConsumer !== undefined &&
+        seed.dailyCallsTotal !== undefined
+      ) {
+        assertConsistentDailyCalls(seed);
+      }
+
+      const updatedVersion: EServiceTemplateVersion = {
+        ...eserviceTemplateVersion,
+        ...(seed.agreementApprovalPolicy !== undefined && {
+          agreementApprovalPolicy:
+            seed.agreementApprovalPolicy === "AUTOMATIC"
+              ? agreementApprovalPolicy.automatic
+              : agreementApprovalPolicy.manual,
+        }),
+        ...(seed.dailyCallsPerConsumer !== undefined && {
+          dailyCallsPerConsumer: seed.dailyCallsPerConsumer,
+        }),
+        ...(seed.dailyCallsTotal !== undefined && {
+          dailyCallsTotal: seed.dailyCallsTotal,
+        }),
+        ...(seed.description !== undefined && {
+          description: seed.description,
+        }),
+        ...(seed.voucherLifespan !== undefined && {
+          voucherLifespan: seed.voucherLifespan,
+        }),
+      };
```

Comment:
When unifying this logic into the helper function, see comment above, here you should implement a logic like the one already implemented for the update of the template for optional fields, see [here](https://github.com/pagopa/interop-be-monorepo/blob/5d781d94d8e9ff4c1f100a9e189a28c4098a5954/packages/eservice-template-process/src/services/eserviceTemplateService.ts#L1924):
- if the field is undefined and it's the POST call, then we need to set it to undefined
- if the field is undefined and it's the PATCH call, then we need to leave it as it is